### PR TITLE
Added Dracula Theme

### LINKED ISSUE #1715: Feature Request: Create a Dracula Theme
Create a Theme based on the popular Dracula dark theme (https://draculatheme.com/).

The theme should use the following color pallet:

image
(https://draculatheme.com/contribute)

### PR DESCRIPTION
**Pull request by codysechelski**
• Added Light and Dark versions of the logo using Dracula Colors
• Added the base theme and Dracula accent colors to the theme.css
• Added the options for the new theme and colors in the theme-settings comp
• Added translations for the theme and color names to all local yaml files that
  already had theme and color information

---
Add a Dracula Theme
---

**Important note**
We may remove your pull request if you do not use this provided PR template correctly.

**Pull Request Type**
Please select what type of pull request this is:
- [ ] Bugfix
- [x] Feature Implementation

**Related issue**
Please link the issue your pull request is referring to. If this pull request fully resolves the relevant issue, put "closes" before the issue number. Example: "closes #123456".
closes #1715 

**Description**
Adds a new Base Theme (Dracula) and several new accent colors based on the popular Dracula Theme (https://draculatheme.com/))

**Screenshots (if appropriate)**
Please add before and after screenshots if there is a visible change.
image
image

**Testing (for code that is not small enough to be easily understandable)**
Has this pull request been tested?
Please describe shortly how you tested it and whether there are any ramifications remaining. 

Manually tested by:
* selecting the new theme and observing the results
* selecting each new color as the main theme color and observing the results
* selecting each new color as the secondary color and observing the results

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] : Windows 10 Home
 - OS Version: [e.g. 22]: Version 21H1, Build 19043.1237 with Windows Feature Experience Pack 120.2212.3530.0
 - FreeTube version: [e.g. 0.8] 0.14.0

**Additional context**

### CODE REVIEW

**Comment by codysechelski** [comment]
PS, This is my first PR to any open source project. I hope I did everything according to spec. 😀

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I didn't tested this yet but damn u did one heck of a job here. 

Not sure but i put it here anyway maybe https://github.com/FreeTubeApp/FreeTube/pull/1482 needs to be merged first and then this PR.

**Comment by jasonhenriquez** [comment]
Also it seems that the manual adding of translations has accidentally erased some translations, mainly "External Link Handling". You will have to fix this @codysechelski. Outside of that, looking from a distance, it looks like some good work!

**Comment by codysechelski** [comment]
Thanks, @efb4f5ff-1298-471a-8973-3d47447115dc!

@jasonhenriquez, Thanks for the feedback I will go back and fix that. Just for my knowledge, going forward, is there a better way to add the translations than manually copying and pasting?

**Comment by jasonhenriquez** [comment]
> @jasonhenriquez, Thanks for the feedback I will go back and fix that. Just for my knowledge, going forward, is there a better way to add the translations than manually copying and pasting?

[Weblate](https://hosted.weblate.org/engage/free-tube/) is usually the preferred tool for translations, but it can be done with pull requests as well. The downside to the latter method is odd merge issues like this, but it is of course still fully welcome as long as all issues are resolved properly.

**Comment by peepopoggers** [comment]
Looks good! Maybe some ideas;
Would it look nicer with the background as the darker section? I am unsure about the darker section being the content area.

**Comment by codysechelski** [comment]
Ok, I think I fixed all of the yaml files that I broke.

**Comment by jasonhenriquez** [comment]
It seems also that your code editor has added two spaces before many of the lines in themes.css, which also should be rectified.

**Comment by codysechelski** [comment]
Cool, I removed the package-lock file. I also tweaked the colors just a fit because I found that the contrast between the background color and the text was a little low in the overflow menus. I also made the cards slightly lighter than the background versus slightly darker. Thanks @peepopoggers for the suggestion.

**Comment by codysechelski** [comment]
Ok so now prettier decided to lowercase all the hex codes and convert double quotes to single quotes. in the theme.css and I didn't notice until after I pushed it. Can I get a ruling on this? Half of me thinks I should revert it as I modified a lot of lines that were not really part of my change. The other half of me thinks that this is an improvement. Most developers I know prefer lower case as it's easier to type.

Please let me know if I should revert it.
Thanks!

**Comment by codysechelski** [comment]
> > @jasonhenriquez, Thanks for the feedback I will go back and fix that. Just for my knowledge, going forward, is there a better way to add the translations than manually copying and pasting?
> 
> [Weblate](https://hosted.weblate.org/engage/free-tube/) is usually the preferred tool for translations, but it can be done with pull requests as well. The downside to the latter method is odd merge issues like this, but it is of course still fully welcome as long as all issues are resolved properly.

So adding the string manually was quite tedious and painful, mainly because I did not trust my merger tool to handle all of the different character sets. I'm interested in how the workflow would be using weblate.

Maybe I just don't understand enough about the underlying code that gets the strings from the yaml files, but as I see it, if you are going to add a string to a screen, then you will reference that string in the yaml files. Otherwise, it will just display the notation.

So do I just need to add the new labels to like the en-us yaml to test that is it working, and then go into weblate and add all of the translation? If so, do I need to wait until after the code is merged to add the translations?

Sorry, I've never really worked much with internationalizing an app and I've never even heard of weblate before now and I'm eager to learn 😀.

If anyone has any documentation on this, please share

**Comment by codysechelski** [comment]
Thanks for the feedback @jasonhenriquez. Yeah, those points make a lot of sense. I have cleaned up the theme.css file and from all I can see in the diff, the only changes in the PR now are ones that directly support this feature.

Please let me know if you have any more feedback for me.

**Comment by peepopoggers** [comment]
This is great but it think the themes dropdown is a bit overwhelming. I can't think of any ideas of howto simplify the list though. Any thoughts?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> 
> 
> This is really good, but now it think the themes dropdown is a bit overwhelming. I can't think of any ideas of howto simplify the list though. Any thoughts?

Maybe only show Dracula colors when Dracula theme is enabled.
After writing this i feel like its not a good thing to do because maybe users like some colors but maybe not the theme.

**Comment by jasonhenriquez** [comment]
> This is really good, but now it think the themes dropdown is a bit overwhelming. I can't think of any ideas of howto simplify the list though. Any thoughts?

I think this aspect alone is fine. However, it may be a factor to consider when reworking the themes. Why would we need to do that, you might ask?

One major issue that FreeTube currently has is inaccessible color contrast with many of the themes. [This is a major accessibility issue for disabled people, particularly Blind and low vision people.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast) 

Just looking at `<ft-button>` elements (e.g., under Privacy Settings), all of the Dracula main color themes have too low color contrast. Surprisingly, however, all of the Dracula secondary color themes have sufficient color contrast. The reason for this difference is because the primary Dracula color themes make text a pale color that is hard to see against many other background colors. This should be an easy fix.

The overarching Dracula theme also adjusts much of the text in FreeTube (e.g., the author & number of views beneath a video on the Subscriptions page) to a similarly ghastly color that is hard to see against the background. This should also be an easy fix.

**Comment by peepopoggers** [comment]
@jasonhenriquez even if it makes it more accessible, it could negatively impact the theme which needs to be taken into consideration. I understand the theme has changed slightly, but in all areas i do agree that the text color should not be too dark as in the previews.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Im not sure if u aware but u have somee conflicts to resolve :)

**Comment by codysechelski** [comment]
Ok, I fixed the contract for the secondary and tertiary colors in the overall theme. The contrast ratios are now as follows:
| Colors Combos                                  | Contract Ratio |
|-----------------------------------------|----------------|
| card-bg-color with primary-text-color   | 13.36          |
| card-bg-color with secondary-text-color | 9.07           |
| card-bg-color with tertiary-text-color  | 11.64          |

I also increased the contrast for all of the primary Dracula colors. Here's the result:
| Primary Color Style | Contract Ratio |
|---------------------|----------------|
| mainDraculaCyan     | 10.29          |
| mainDraculaGreen    | 10.38          |
| mainDraculaOrange   | 8.36           |
| mainDraculaPink     | 5.97           |
| mainDraculaPurple   | 5.9            |
| mainDraculaRed      | 4.53           |
| mainDraculaYellow   | 12.75          |

**Review by jasonhenriquez — APPROVED** [review]
I vant to let you know how I impressed by not only your vork but also your dedication. Fangk you very much, and cross my heart, it vould be a mis-stake for me to not approve this PR! 🧛‍♂️👍 (Seriously, good job!)

**Comment by codysechelski** [comment]
Thanks, @jasonhenriquez, you are too kind. This was my very first PR to an opensource project and you are the others here have made me fell very welcome. I enjoyed the process I'm looking forward to my next commit! 👍

**Review by peepopoggers — APPROVED** [review]
Tested this out, colors look great! Good job.

File: static/locales/sr.yaml
Comment (ChunkyProgrammer) [inline static/locales/sr.yaml:176]: I'm gonna guess you didn't mean to do this change.

**Review by ChunkyProgrammer — APPROVED** [review]
Code looks great. Thanks so much for taking the time to work on this PR

File: static/locales/sr.yaml
Comment (jasonhenriquez) [reply static/locales/sr.yaml:176]: Good catch. Thankfully, It seems to have just reverted to an older translation for that string, and shouldn't pose that significant of a problem.

**Pull request merged**
